app/controllers/internal/users_controller.rb

```diff
@@ -58,7 +58,7 @@ def full_delete
     @user = User.find(params[:id])
     begin
       Moderator::DeleteUser.call(admin: current_user, user: @user, user_params: user_params)
-      flash[:success] = "@" + @user.username + " (email: " + @user.email + ", user_id: " + @user.id.to_s + ") has been fully deleted. If requested, old content may have been ghostified. If this is a GDPR delete, delete them from Mailchimp & Google Analytics."
+      flash[:success] = "@#{@user.username} (email: #{@user.email || 'no email'}, user_id: #{@user.id}) has been fully deleted. If requested, old content may have been ghostified. If this is a GDPR delete, delete them from Mailchimp & Google Analytics."
```

Comment:
Added a default value of "no email" to make the flash message look nicer.

Reply:
Empty string doesn't work in the OR and we can save a user with an empty email :-/

Reply:
You're correct. Here's the full email validation for reference:

```ruby
  validates :email,
            length: { maximum: 50 },
            email: true,
            allow_nil: true
```

For this specific line `@user.email.presence || 'no email'` will do, I'll address this all tomorrow. Thanks!